src/main/java/com/faforever/api/user/UserController.java

```diff
@@ -77,4 +82,25 @@ public void claimPasswordResetToken(HttpServletResponse response,
     userService.claimPasswordResetToken(token, newPassword);
     response.sendRedirect(fafApiProperties.getPasswordReset().getSuccessRedirectUrl());
   }
+
+  @PreAuthorize("#oauth2.hasScope('write_account_data') and hasRole('ROLE_USER')")
+  @ApiOperation("Creates an URL to the steam platform to initiate the Link To Steam process.")
+  @RequestMapping(path = "/buildSteamLinkUrl", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
+  public Map<String, Serializable> buildSteamLinkUrl(Authentication authentication) {
+    String steamUrl = userService.buildSteamLinkUrl(userService.getUser(authentication));
+    return ImmutableMap.of("steam_url", steamUrl);
+  }
+
+  @ApiOperation("Processes the Steam redirect and creates the steam link in the user account.")
+  @RequestMapping(path = "/linkToSteam", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+  public void linkToSteam(HttpServletRequest request,
+                          HttpServletResponse response,
+                          @RequestParam("token") String token) throws IOException {
+    try {
+      userService.linkToSteam(token, steamService.parseSteamIdFromLoginRedirect(request));
+      response.sendRedirect(fafApiProperties.getLinkToSteam().getSuccessRedirectUrl());
+    } catch (ApiException e) {
+      response.sendRedirect(String.format(fafApiProperties.getLinkToSteam().getErrorRedirectUrlFormat(), e.getMessage()));
```

Comment:
The message of an `ApiException` is not very human readable. We might want to pass an array of error codes instead and render it properly.